File: MicroServices/MicroServices/referenciel-service/src/main/java/com/bfi/referentiel/service/WeeklyService.java
package com.bfi.referentiel.service;

import com.bfi.referentiel.common.dto.WeeklyDTO;

import java.util.List;

public interface WeeklyService {

    void addEWeekly(WeeklyDTO weekly);

    void updateWeekly(WeeklyDTO weekly);

    void deleteWeekly(Long id);

    List<WeeklyDTO> getAllWeeklys();

    List<WeeklyDTO> findWeeklysByCalendarId(Long id);
}

File: CartagoFrontV2-master/src/app/services/referenciel/weekly.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Weekly } from '../../models/weekly';

@Injectable({
  providedIn: 'root',
})
export class WeeklyService {
  constructor(private http: HttpClient) {}

  getAllWeeklys(): Observable<Weekly[]> {
    return this.http.get<Weekly[]>('http://localhost:8080/Weeklys');
  }

  addWeekly(weekly: Weekly): Observable<Weekly> {
    return this.http.post<Weekly>('http://localhost:8080/Weeklys/', weekly);
  }

  updateWeekly(weekly: Weekly): Observable<Weekly> {
    return this.http.put<Weekly>('http://localhost:8080/Weeklys/', weekly);
  }

  deleteWeekly(id: number): Observable<Weekly> {
    return this.http.delete<Weekly>('http://localhost:8080/Weeklys/' + id);
  }
}
